book/ecosystem.ipynb

```diff
@@ -0,0 +1,259 @@
+   "outputs": [],
+   "source": [
+    "import pygmt\n",
+    "import geopandas as gpd"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a0f88acb-e463-4193-a7ef-33837b2f5fdf",
+   "metadata": {},
+   "source": [
+    "We'll load sample data provided through the GeoPandas package and inspect the GeoDataFrame."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8cf79e91-8ce0-48ec-8812-1395d3d0eddf",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "world = gpd.read_file(gpd.datasets.get_path('naturalearth_lowres'))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6a44f98e-5b50-463f-829a-cd0fa25a6aa1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "world.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "65adae6a-3933-453b-bd4b-4e04e018d02d",
+   "metadata": {},
+   "source": [
+    "Following the [GeoPandas example](https://geopandas.org/en/stable/docs/user_guide/mapping.html#choropleth-maps), we'll create a Choropleth map showing the GDP per capita, but will use PyGMT to plot the data using the [Hammer projection](https://www.pygmt.org/latest/projections/misc/misc_hammer.html#hammer)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "76b850c4-e5bc-42bd-ba13-7b52b4f60dc2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Calculate the GDP per capita\n",
+    "world = world[(world.pop_est>0) & (world.name!=\"Antarctica\")]\n",
+    "world['gdp_per_cap'] = world.gdp_md_est / world.pop_est\n",
+    "\n",
+    "# Find the range of data values for creating a colormap\n",
+    "cmap_bounds = pygmt.info(data=world['gdp_per_cap'], per_column=True)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "864091e5-85e2-4452-ab44-5cbeaa2ac8a9",
+   "metadata": {},
+   "source": [
+    "Now, we'll plot the data on a PyGMT figure, by creating a figure instance, laying down a basemap, plotting the GeoDataFrame, and adding a colorbar!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "15a21a74-8531-4e74-86e1-a85e6eaef793",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create an instance of the pygmt.Figure class\n",
+    "fig = pygmt.Figure()\n",
+    "# Create a colormap for the figure\n",
+    "pygmt.makecpt(cmap=\"bilbao\", series=cmap_bounds)\n",
+    "# Create a basemap\n",
+    "fig.basemap(region=\"g\", projection=\"H15c\", frame=True)\n",
+    "# Plot the GeoDataFrame\n",
+    "# - Use `close=True` to specify that the polygons should be forced closed\n",
+    "# - Plot the polygon outlines with a 1 point, black pen\n",
+    "# - Set that the color should be based on the `gdb_per_cap` using the `color, `cmap`, and `aspatial` parameters\n",
+    "fig.plot(data=world, pen=\"1p,black\", close=True, color=\"+z\", cmap=True, aspatial=\"Z=gdp_per_cap\")\n",
+    "# Add a colorbar\n",
+    "fig.colorbar(position=\"JMR\", frame=\"a0.02\")\n",
```

Comment:
Some units for the colorbar would be nice here. I don't understand the number range though, it goes from 0 up to 0.14, is GDP per capita meant to be that low?

Reply:
I'll update it to plot the world population column instead for simplicity.